lib/ClientConnectContext.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Amp\Socket;
+
+use function Amp\Socket\Internal\normalizeBindToOption;
+
+final class ClientConnectContext {
+    private $bindTo = null;
+    private $connectTimeout = 10000;
+    private $maxAttempts = 2;
```

Comment:
Not sure what's a sane default here. A name resolving to multiple IPs will count each IP as an attempt.